src/ui/components/ProtocolViewer/suspense/recordedProtocolMessagesCache.ts

```diff
@@ -86,10 +86,9 @@ export const recordedProtocolMessagesCache = createFocusIntervalCacheForExecutio
           return [];
         }
 
-        const preferredLocation = getPreferredLocation(
-          sourcesState,
-          pointDescriptions[0].frame ?? []
-        );
+        const preferredLocation =
+          pointDescriptions[0].frame &&
+          getPreferredLocation(sourcesState, pointDescriptions[0].frame);
```

Comment:
I wonder if it would be easier/safer for us to update `getPreferredLocation` to support either `MappedLocation | undefined` as the third parameter?

Reply:
I don't have a strong preference. I can imagine us adding a new callsite and not knowing to avoid the `?? []` pattern here is the only reason I asked. Probably fine.